packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -181,19 +210,31 @@ function nodeDataToMapTree(
 
 	const schema = getType(data, allowedTypes);
 
+	if (allowNonEnabledTypes === false) {
+		const annotations = annotatedAllowedTypes.get(schema);
+		if (annotations !== undefined) {
+			if (annotations.enabledUponSchemaUpgrade !== undefined) {
+				// TODO:#38722 check if the upgrade to enablable has happened, once the upgrade mechanism is implemented
+				throw new UsageError("Schema type is not enabled for this context");
```

Comment:
We should make this error more friendly (maybe include the type that's not allowed, or if we make the enabledUponSchemaUpgrade something more useful, derive the message from it)